### ISSUE TITLE
gcc compiler is not supported

### ISSUE THREAD

**Issue by JakubOnderka** [gyscos/zstd-rs#51]
Hi, I tried to compile zstd-rs in Docker container with latest rust and build-essential package, that's contains gcc compiler:

```
$ rustc -v --version      
rustc 1.30.1 (1433507eb 2018-11-07)
binary: rustc
commit-hash: 1433507eba7d1a114e4c6f27ae0e1a74f60f20de
commit-date: 2018-11-07
host: x86_64-unknown-linux-gnu
release: 1.30.1
LLVM version: 8.0
```

``` 
$ gcc --version
gcc (Ubuntu 7.3.0-27ubuntu1~18.04) 7.3.0
```

but with gcc it is not possible to compile this crate, because of this error: 

```
thread 'main' panicked at 'Unable to find libclang: "couldn\'t find any valid shared libraries matching: [\'libclang.so\', \'libclang-*.so\', \'libclang.so.*\'], set the `LIBCLANG_PATH` environment variable to a path where one of these files can be found (invalid: [])"', libcore/result.rs:1009:5
note: Run with `RUST_BACKTRACE=1` for a backtrace.
```

so it looks like clang is required to compile this crate but at least there is no information that clang is required.

**Comment by gyscos** [comment]
Using the `rust:latest` image, I can successfully clone this repo and run `cargo test`.
What image are you starting from? What command are you running when you get this error?

**Comment by gyscos** [comment]
It seems this was using gcc 6.3.0

Using `ubuntu:latest` with gcc-7.3.0, runing:
```
$ apt update && apt install git curl build-essential && curl https://sh.rustup.rs -sSf | sh
$ cd ~ && source .profile
$ git clone https://github.com/gyscos/zstd-rs && cd zstd-rs
$ git submodule update --recursive --init
$ cargo test
```
works fine too.

**Comment by JakubOnderka** [comment]
I was trying Ubuntu 18.04 (latest LTS). But current master works fine, so the problem is just in latest released version (zstd = "0.4.14")

**Comment by jhgg** [comment]
We can reproduce this issue too on Windows. Using the following pinned versions continues to work:

```toml

[[package]]
name = "zstd"
version = "0.4.19+zstd.1.3.5"
source = "registry+https://github.com/rust-lang/crates.io-index"
dependencies = [
 "libc 0.2.43 (registry+https://github.com/rust-lang/crates.io-index)",
 "zstd-safe 1.4.4+zstd.1.3.5 (registry+https://github.com/rust-lang/crates.io-index)",
]

[[package]]
name = "zstd-safe"
version = "1.4.4+zstd.1.3.5"
source = "registry+https://github.com/rust-lang/crates.io-index"
dependencies = [
 "libc 0.2.43 (registry+https://github.com/rust-lang/crates.io-index)",
 "zstd-sys 1.4.4+zstd.1.3.5 (registry+https://github.com/rust-lang/crates.io-index)",
]

[[package]]
name = "zstd-sys"
version = "1.4.4+zstd.1.3.5"
source = "registry+https://github.com/rust-lang/crates.io-index"
dependencies = [
 "blob 0.2.0 (registry+https://github.com/rust-lang/crates.io-index)",
 "gcc 0.3.54 (registry+https://github.com/rust-lang/crates.io-index)",
 "glob 0.2.11 (registry+https://github.com/rust-lang/crates.io-index)",
 "libc 0.2.43 (registry+https://github.com/rust-lang/crates.io-index)",
]
```

However,
```toml
[[package]]
name = "zstd"
version = "0.4.21+zstd.1.3.7"
source = "registry+https://github.com/rust-lang/crates.io-index"
dependencies = [
 "zstd-safe 1.4.6+zstd.1.3.7 (registry+https://github.com/rust-lang/crates.io-index)",
]

[[package]]
name = "zstd-safe"
version = "1.4.6+zstd.1.3.7"
source = "registry+https://github.com/rust-lang/crates.io-index"
dependencies = [
 "libc 0.2.44 (registry+https://github.com/rust-lang/crates.io-index)",
 "zstd-sys 1.4.6+zstd.1.3.7 (registry+https://github.com/rust-lang/crates.io-index)",
]

[[package]]
name = "zstd-sys"
version = "1.4.6+zstd.1.3.7"
source = "registry+https://github.com/rust-lang/crates.io-index"
dependencies = [
 "bindgen 0.42.3 (registry+https://github.com/rust-lang/crates.io-index)",
 "blob 0.2.0 (registry+https://github.com/rust-lang/crates.io-index)",
 "cc 1.0.25 (registry+https://github.com/rust-lang/crates.io-index)",
 "glob 0.2.11 (registry+https://github.com/rust-lang/crates.io-index)",
 "libc 0.2.44 (registry+https://github.com/rust-lang/crates.io-index)",
]
``` 

Does not, for the exact same error. This is using:

```
rustup show
Default host: x86_64-pc-windows-msvc

installed toolchains
--------------------

stable-i686-pc-windows-msvc
stable-x86_64-pc-windows-gnu
stable-x86_64-pc-windows-msvc
nightly-x86_64-pc-windows-msvc
1.29.0-i686-pc-windows-msvc
1.29.0-x86_64-pc-windows-msvc
1.30.0-i686-pc-windows-msvc
1.30.0-x86_64-pc-windows-msvc (default)

installed targets for active toolchain
--------------------------------------

i686-pc-windows-msvc
x86_64-pc-windows-msvc

active toolchain
----------------

1.30.0-x86_64-pc-windows-msvc (default)
rustc 1.30.0 (da5f414c2 2018-10-24)
```

**Comment by gyscos** [comment]
Interesting, the problem could be the move from `gcc` to the `cc` crate? They should work pretty similarly.
Does the current git master works? And what version of gcc are you using?

**Comment by jonathanstrong** [comment]
encountered same issue on ubuntu 16.04. switching to the 0.4.19/1.4.4 versions fixed it for me as well.

```console
$ gcc --version
gcc (Ubuntu 5.4.0-6ubuntu1~16.04.10) 5.4.0 20160609

$ rustc --version
rustc 1.32.0-nightly (b3af09205 2018-12-04)
```

**Comment by bruceg** [comment]
I encountered the same issue on Gentoo, switching to the 0.4.19/1.4.4 versions fixed it for me as well.
```
> gcc --version
gcc (Gentoo 7.4.0 p1.1) 7.4.0

> rustc --version
rustc 1.31.1 (b6c32da9b 2018-12-18)
```

I bisected this down to commit 6e9fc3aa392dab604da2658ddf53127be097dee1, and confirmed that it is correct by reverting it on top of 0.4.21. However, The bug is not present in the master branch, so a release of the current sources should resolve this issue.
